3 implement fileuploader service for image and cv storage via supabase buckets

NKSurprise opened the pull request.

iliyancherganski committed:
Add user profile creation endpoint and validation

Introduces CreateUserProfileRequest model, validator, and service method to support user profile creation. Adds a POST endpoint to UserProfileController for creating profiles, centralizes validation logic in ValidationConstants, and refactors update validator to use shared validation. Updates file handling in UserProfileService for profile creation and deletion.

NKSurprise committed:
Change Create action to accept form data

Updated the Create method in UserProfileController to use [FromForm] instead of [FromBody] for CreateUserProfileRequest. This allows the endpoint to accept form data submissions.

dpS1lence commented:
@NKSurprise Това тествано ли е с файлове през постман?

iliyancherganski committed:
Add project assignment search with specifications

Introduces a flexible search endpoint for project assignments using the specification pattern. Adds request and specification models for filtering by title, level, deadline, company name, sector, and skills. Updates the controller and service interfaces to support searching with pagination (to be implemented), and provides the core logic for combining and evaluating specifications.

iliyancherganski committed:
Merge branch 'dev' into 3-implement-fileuploader-service-for-image-and-cv-storage-via-supabase-buckets

ImSk1 commented on src/SkillBridge/Services/ProjectAssignment/IProjectAssignmentService.cs:
```diff
@@ -100,6 +101,15 @@ public interface IProjectAssignmentService
+
+    // Change when PagedList is implemented
+    // public async Task<PagedList<ProjectAssignmentResponse>> SearchProjectAssignmentsAsync
+    Task<IEnumerable<ProjectAssignmentResponse>> SearchProjectAssignmentsAsync(Specification<Models.Entities.ProjectAssignment> specification,
```
Rename to SearchAsync

ImSk1 commented on src/SkillBridge/Services/UserProfile/UserProfileService.cs:
```diff
@@ -39,6 +40,41 @@ public UserProfileService(
+            var auth0UserId = userId ?? _currentUser.GetUserId();
+            _logger.LogInformation("Retrieving profile with ID: {UserProfileId}", auth0UserId);
+
+            if (await _dbContext.UserProfiles.FindAsync(auth0UserId) == null)
```
Reverse if statement so it doesn't contain the whole method.

iliyancherganski committed:
Refactor project assignment search and fix user profile creation

Renamed SearchProjectAssignmentsAsync to SearchAsync in the project assignment service and updated all references for consistency. Fixed logic in UserProfileService to prevent creating a user profile if one already exists, throwing an exception instead.

iliyancherganski committed:
Revert "Merge branch 'dev' into 3-implement-fileuploader-service-for-image-and-cv-storage-via-supabase-buckets"

This reverts commit 38b9878dd10a25bfdc570448ca19f692aa7e38d1, reversing
changes made to f72e751f4677ce8f85aefa7d2287eeb251da0a26.
